src/App.tsx

```diff
@@ -8,53 +8,136 @@
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getData } from './api/fetch';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userId, setUserId] = useState<number | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [isLoading, setIsLoading] = useState(false);
+  const [catchError, setCatchError] = useState(false);
 
-              <Loader />
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedpost] = useState<Post | null>(null);
+  const [emptyPosts, setEmptyPosts] = useState(false);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const fetchUsers = async () => {
+    try {
+      setIsLoading(true);
+      const url = '/users';
+      const currentUsers = await getData<User>(url);
+
+      setUsers(currentUsers);
+    } catch {
+      setCatchError(true);
+    } finally {
+      setIsLoading(false);
+      setTimeout(() => {
+        setCatchError(false);
+      }, 3000);
+    }
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
+  useEffect(() => {
+    setPosts([]);
+
+    if (users && userId) {
+      const fetchPosts = async () => {
+        try {
+          setIsLoading(true);
+          const url = `/posts?userId=${userId}`;
+
+          if (userId) {
+            const currentPosts = await getData<Post>(url);
+
+            if (currentPosts.length === 0) {
+              setEmptyPosts(true);
+            }
+
+            setPosts(currentPosts);
+          }
+        } catch {
+          setCatchError(true);
+        } finally {
+          setIsLoading(false);
+        }
+      };
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+      fetchPosts();
+    }
+  }, [users, userId]);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  userId={userId}
+                  setUserId={setUserId}
+                  setSelectedpost={setSelectedpost}
```

Comment:
The prop `setSelectedpost` should be renamed to `setSelectedPost` to maintain consistency with the state variable naming.